### PR TITLE
[FEATURE] Transfer Ownership of the Channel

### LINKED ISSUE #1864: [FEATURE] Transfer Ownership of the Channel
- Your Rocket.Chat Experimental app version: 4.5.1.9999999
- Your Rocket.Chat server version: 3.1.0-develop
 Device (or Simulator) you're running with: Honor 9 lite

There is no option for transferring ownership of the channel to another member of the channel.

### PR DESCRIPTION
**Pull request by devyaniChoubey**
@diegolmello I have to add the feature of 'Ignore a User' also in this pr. But please review these changes first and please tell me whether this is the correct way to fix the issue.
@RocketChat/ReactNative

Closes #1864

### CODE REVIEW

**Comment by devyaniChoubey** [comment]
@diegolmello please review

File: app/lib/rocketchat.js [hunk 1]
```diff
@@ -1000,6 +1000,31 @@ const RocketChat = {
 			searchText
 		});
 	},
+	toggleOwner(roomId, userId, owner, t) {
+		// RC 1.0
```
Comment (diegolmello) [inline app/lib/rocketchat.js]: Where did you get this server version from?
Take a look at our docs: https://rocket.chat/docs/developer-guides/rest-api/channels/addowner/#change-log
It was introduced on version 0.49.4.

File: app/views/RoomMembersView/index.js [hunk 2]
```diff
@@ -90,13 +90,17 @@ class RoomMembersView extends React.Component {
...
 
 		const { navigation } = this.props;
-		const { rid } = navigation.state.params;
 		navigation.setParams({ toggleStatus: this.toggleStatus });
-		this.permissions = await RocketChat.hasPermission(['mute-user'], rid);
+		this.SET_OWNER_INDEX = this.actionSheetOptions.length - 1;
+		this.SET_LEADER_INDEX = this.actionSheetOptions.length - 1;
+		this.SET_MODERATOR_INDEX = this.actionSheetOptions.length - 1;
+		this.MUTE_INDEX = this.actionSheetOptions.length - 1;
+		this.REMOVE_USER_INDEX = this.actionSheetOptions.length - 1;
```
Comment (diegolmello) [inline app/views/RoomMembersView/index.js]: I didn't get this. Can you elaborate?

File: app/views/RoomMembersView/index.js [hunk 3]
```diff
@@ -134,16 +138,71 @@ class RoomMembersView extends React.Component {
...
-			return;
+		if (!this.muteUserPermission) {
+			return false;
 		}
 		const { room } = this.state;
-		const { muted } = room;
-
+		this.isLeader = room && room.roles && room.roles.length && !!room.roles.find(role => role === 'leader');
+		this.isOwner = room && room.roles && room.roles.length && !!room.roles.find(role => role === 'owner');
+		this.isModerator = room && room.roles && room.roles.length && !!room.roles.find(role => role === 'moderator');
```
Comment (diegolmello) [inline app/views/RoomMembersView/index.js]: `leader`, `owner` and `moderator` could be constants, as you're going to use them below.

File: app/lib/rocketchat.js (hunk 1 above)
Comment (devyaniChoubey) [reply app/lib/rocketchat.js]: @diegolmello I have gone through the docs to get the api endpoints but I forgot to change server version. Sorry for that

File: app/views/RoomMembersView/index.js (hunk 2 above)
Comment (devyaniChoubey) [reply app/views/RoomMembersView/index.js]: these are actionIndex

**Comment by devyaniChoubey** [comment]
@diegolmello  @djorkaeffalexandre please review

**Comment by devyaniChoubey** [comment]
REMOVE USER 
GIF-200328_161832

**Comment by devyaniChoubey** [comment]
REMOVE USER
GIF-200328_171016

**Pull request closed without merging**